packages/effect/src/Array.ts

```diff
@@ -119,8 +119,11 @@ export const replicate: {
  * @category constructors
  * @since 2.0.0
  */
-export const fromIterable = <A>(collection: Iterable<A>): Array<A> =>
-  Array.isArray(collection) ? collection : Array.from(collection)
+export function fromIterable<A, B>(collection: Iterable<A>, f: (a: A, i: number) => B): Array<B>
+export function fromIterable<A>(collection: Iterable<A>): Array<A>
+export function fromIterable<A>(collection: Iterable<A>, f?: (a: A, i: number) => A): Array<A> {
+  return f ? Array.from(collection, f) : Array.isArray(collection) ? collection : Array.from(collection)
+}
```

Comment:
This is imho a breaking change because it add a new position parameter that can cause problems with tacit usage.

Reply:
Yes, I haven't changed the test to show that it's a breaking change, I think we should close this and tag the corresponding issue with a "next-major" label.

Reply:
Agreed

Reply:
I don't think we generally support tacit usage